cmd/nomos/status/client.go

```diff
@@ -255,10 +255,16 @@ func (c *ClusterClient) multiRepoClusterStatus(ctx context.Context, cs *ClusterS
 		}
 	}
 
-	if len(rootSyncs) != 0 {
+	if len(rootSyncs) != len(rootRGs) {
+		errs = append(errs, fmt.Sprintf("expected the number of RootSyncs and ResourceGroups to be equal, but found %d RootSyncs and %d ResourceGroups", len(rootSyncs), len(rootRGs)))
+	} else if len(rootSyncs) != 0 {
 		var repos []*RepoState
 		for i, rs := range rootSyncs {
 			rg := rootRGs[i]
+			if rg == nil {
+				// We always expect a ResourceGroup, even if we have no managed resources.
+				errs = append(errs, fmt.Sprintf("resourcegroups.kpt.dev %q not found", rs.Name))
```

Comment:
Can you add namespace to the error so that we know if that's for root-sync or repo-sync, and all other resourcegroups as well?

Reply:
Done.